src/Tokenizer.php

```diff
@@ -862,7 +862,7 @@ private function createNextToken(string $string, Token|null $previous = null): T
                 $value = $string[0] . $this->getNextQuotedString(substr($string, 1));
             } else {
                 // Non-quoted variable name
-                preg_match('/^(' . $string[0] . '[a-zA-Z0-9\._\$]+)/', $string, $matches);
+                preg_match('/^(' . $string[0] . '[\w.$]+)/', $string, $matches);
```

Comment:
Can you elaborate? `.` will match anything, won't it?

Reply:
Ah no it seems fine when inside a character class.

Reply:
yes - we are in regex class